call.go

```diff
@@ -61,7 +61,7 @@ func recvResponse(ctx context.Context, dopts dialOptions, t transport.ClientTran
 		if c.maxReceiveMessageSize == nil {
 			return Errorf(codes.Internal, "callInfo maxReceiveMessageSize field uninitialized(nil)")
 		}
-		if err = recv(p, dopts.codec, stream, dopts.dc, reply, *c.maxReceiveMessageSize, inPayload); err != nil {
+		if err = recv(p, dopts.codec, stream, dopts.dc, reply, *c.maxReceiveMessageSize, inPayload, encoding.GetCompressor(c.compressorType)); err != nil {
```

Comment:
With this change, if the user does `UseCompressor("something that does not exist")`, no error will ever be returned, we will just silently ignore that `CallOption`.  We should make it so when this happens an error (probably `Internal`) is returned from the RPC (early).